src/thingset_bin.c

```diff
@@ -141,6 +161,44 @@ static int bin_serialize_path(struct thingset_context *ts,
                : -THINGSET_ERR_RESPONSE_TOO_LARGE;
 }
 
+static int bin_serialize_metadata(struct thingset_context *ts,
+                                  const struct thingset_data_object *object)
+{
+    int err = bin_serialize_map_start(ts);
+    if (err) {
+        return err;
+    }
+
+    const char *name = "name";
+    if (!zcbor_tstr_encode_ptr(ts->encoder, name, 4)) {
+        return -THINGSET_ERR_RESPONSE_TOO_LARGE;
+    }
+
+    if (!zcbor_tstr_encode_ptr(ts->encoder, object->name, strlen(object->name))) {
+        return -THINGSET_ERR_RESPONSE_TOO_LARGE;
+    }
+
+    const char *type = "type";
+    if (!zcbor_tstr_encode_ptr(ts->encoder, type, 4)) {
+        return -THINGSET_ERR_RESPONSE_TOO_LARGE;
+    }
+
+    char buf[128];
+    int len = thingset_get_type_name(ts, object, (char *)&buf, sizeof(buf));
+    if (len < 0) {
+        return -1;
```

Comment:
-1 is not a valid ThingSet error code, I think. What about this?
```suggestion
        return -THINGSET_ERR_RESPONSE_TOO_LARGE;
```